src/lightning_habana/pytorch/datamodule/utils.py

```diff
@@ -0,0 +1,68 @@
+# Copyright (c) 2023 Habana Labs, Ltd. an Intel Company
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from lightning_habana.utils.imports import _HPU_AVAILABLE
+
+if _HPU_AVAILABLE:
+    import habana_frameworks.torch.hpu as torch_hpu
+
+import torch.distributed as dist
+
+
+def is_dist_avail_and_initialized() -> bool:
+    """Check distributed backend is initialized or not."""
+    if not dist.is_available():
+        return False
+    if not dist.is_initialized():
+        return False
+    return True
```

Comment:
Since recent PyTorch versions, you can just check `is_initialized` and it will return False if it's not available

Reply:
Thanks, wasn't  aware of it. Will try out in another PR